banyand/liaison/grpc/grpc.go

```diff
@@ -128,3 +210,92 @@ func (s *Server) GracefulStop() {
 	s.log.Info().Msg("stopping")
 	s.ser.GracefulStop()
 }
+
+func assemblyWriteData(shardID uint, writeEntity *v1.WriteRequest, seriesID uint64) data.TraceWriteDate {
+	return data.TraceWriteDate{ShardID: shardID, SeriesID: seriesID, WriteRequest: writeEntity}
+}
+
+func (s *Server) Write(TraceWriteServer v1.TraceService_WriteServer) error {
+	for {
+		writeEntity, err := TraceWriteServer.Recv()
+		if err == io.EOF {
+			return nil
+		}
+		if err != nil {
+			return err
+		}
+
+		ana := logical.DefaultAnalyzer()
+		metadata := common.Metadata{
+			KindVersion: apischema.SeriesKindVersion,
+			Spec:        writeEntity.GetMetadata(),
+		}
+		schema, ruleError := ana.BuildTraceSchema(context.TODO(), metadata)
+		if ruleError != nil {
+			return ruleError
+		}
+		s.seriesInfo.RWMutex.RLock()
+		if s.seriesInfo.seriesEvent == nil {
+			s.seriesInfo.RWMutex.RUnlock()
```

Comment:
How about adding a `s.getSeriesInfo` here like 

```go
func (s *Server) getSeriesInfo() {
    s.RWMutex.RLock()
    defer s.RWMutext.RUlock()
    return s.xxxx
}
```

The above method helps us get rid of checking where to unlock rwmutext.